packages/common/src/rule-engine/rules/schemas.ts

```diff
@@ -0,0 +1,23 @@
+import { z } from 'zod';
+
+import { OPERATION, OPERATOR } from '../operators/enums';
+import { OperationsValueSchema } from '../operators/constants';
+import { RuleSet } from './types';
+import { AmlCheckSchema } from '../operators/schemas';
+
+export function getValues<T extends Record<string, unknown>>(obj: T) {
+  return Object.values(obj) as [(typeof obj)[keyof T]];
+}
+
+export const RuleSchema = z
+  .object({
+    key: z.string(),
+    operator: z.enum(getValues(OPERATION)),
+  })
+  .and(OperationsValueSchema);
+
+// @ts-ignore - cycle zod types are not correct
```

Comment:
There's a way to use Zod's lazy schemas, you didn't use it well. Please refer to the docs and fix it.

Reply:
couldnt find a reference for that, its looks find as they mentioned in their docs

Reply:
If you had use it as they mentioned in the docs, you wouldn't have to use `@ts-ignore`: https://zod.dev/?id=recursive-types